pmd-core/src/main/java/net/sourceforge/pmd/ant/Formatter.java

```diff
@@ -57,12 +61,26 @@ public Renderer getRenderer() {
     }
 
     public void start(String baseDir) {
+
+        Properties properties = createProperties();
+        
+        Charset charset;
+        {
+            String s = (String) properties.get("encoding");
+            if (null == s) {
+                charset = StandardCharsets.UTF_8;
+                properties.put("encoding", charset.name());
+            } else {
+                charset = Charset.forName(s);
+            }
+        }
+
         try {
             if (toConsole) {
-                writer = new BufferedWriter(new OutputStreamWriter(System.out));
+                writer = new BufferedWriter(new OutputStreamWriter(System.out, charset));
```

Comment:
Should the console should always use the platform encoding? I would expect this, as otherwise you might see strange characters. However, if you redirect the output to a file, then we should use the specified encoding...

See also http://stackoverflow.com/questions/1403772/how-can-i-check-if-a-java-programs-input-output-streams-are-connected-to-a-term - we could use [System.console()](http://docs.oracle.com/javase/6/docs/api/java/lang/System.html#console()) for a simple check.